include/envoy/http/filter.h

```diff
@@ -151,6 +151,19 @@ class StreamFilterCallbacks {
    */
   virtual Router::RouteConstSharedPtr route() PURE;
 
+  /**
+   * Invokes callback with a matched route, callback can choose to accept this route by returning
+   * Router::RouteMatchStatus::Accept or continue route match from last matched route by returning
+   * Router::RouteMatchStatus::Continue, if there are more routes available.
+   *
+   * Returns route accepted by the callback or nullptr if no match found or none of route is
+   * accepted by the callback.
+   *
+   * NOTE: clearRouteCache() must be called before invoking this method otherwise cached route will
```

Comment:
To be fair to @kprakasam this is how it was originally written, but I had him change it to be consistent with how the `route()` method works. IMO this consistency is easier to understand, but if you feel otherwise happy to revisit.

Reply:
I think its a bit unclear how `route()` works with the existing description/name, I've definitely written code like

```
<modify headers>
callbacks_.clearRouteCache();
// This re-resolves the route.
callbacks_.route();
```
because `route()` looks like a getter function at a glance, with some implications that it will re-resolve the route after a `clearRouteCache` if you read the comment. 

I kinda wish that `route()` was made const and instead we'd have `resolveRoute()` and `resolveRoute(const Router::RouteCallback& cb)` to be super explicit about what everything does, though it might not be worth the churn of this public API? WDYT?

Reply:
I agree what you propose would be optimal, but I would suggest we track that as a separate tech debt follow up and not block @kprakasam on that. I think what we have here is conistent, if imperfect. Thoughts?

Reply:
Yeah that works for me, I can file a separate ticket